.github/workflows/ci.yaml

```diff
@@ -142,25 +142,41 @@ jobs:
     if: github.event_name == 'pull_request'
     steps:
       - uses: actions/checkout@v3
+      - uses: tj-actions/changed-files@v37
+        id: filter
+        with:
+          files: |
+            Cargo.toml
+            Cargo.lock
+            crates/**
+            !crates/ruff_python_formatter/**
+            !crates/ruff_formatter/**
+            !crates/ruff_dev/**
+            !crates/ruff_shrinking/**
+
       - uses: actions/setup-python@v4
+        if: steps.filter.outputs.any_changed == 'true'
```

Comment:
Would a single job that determines if {linter, formatted} changed and triggers downstream accordingly be possible? It would mean an extra job, but we would need only one guard per job and the checks list would show what actually ran

Reply:
I guess that would be nice... argh, I have to re-do all tests

Reply:
Could we do graphite PRs for the different parts on top? I volunteer #5898 as a dev tools PR ;)

Reply:
How do you mean? I now changed it to a job. But I have to re-do all my testing... which gets funny because I now can't link to jobs that didn't run

Reply:
Add PRs on top of this PR that change formatter/linter/dev tools. Graphite can rebase them and you can look at their CI runs

Reply:
Oh, that would have been clever. Next time